code/iaas/config-item/server/src/main/java/io/cattle/platform/configitem/context/data/LoadBalancerTargetsInfo.java

```diff
@@ -11,10 +11,22 @@
     List<LoadBalancerTargetInfo> targets;
     InstanceHealthCheck healthCheck;
 
-    public LoadBalancerTargetsInfo(List<LoadBalancerTargetInfo> lbTargetInfo, InstanceHealthCheck healthCheck) {
+    public LoadBalancerTargetsInfo(List<LoadBalancerTargetInfo> lbTargetInfo, InstanceHealthCheck lbHealthCheck) {
         super();
         this.targets = lbTargetInfo;
-        this.healthCheck = healthCheck;
+        for (LoadBalancerTargetInfo target : this.targets) {
+            if (target.getHealthCheck() != null) {
```

Comment:
yes, in this case we pick up the first one to assign to TargetsInfo. Only one parameter is read from this healtcheck on TargetSInfo level (backend element on haproxy) anyway when it comes to haproxy configuration (this parameter is http requestLine). At the target level (server element on haproxy) healtcheck is read on per target basis (from TargetInfo)